app/data_models/questionnaire_store.py

```diff
@@ -10,18 +12,21 @@
 class QuestionnaireStore:
     LATEST_VERSION = 1
 
-    def __init__(self, storage, version=None):
+    def __init__(self, storage: Any, version: int = None):
+        # Storage type hint needs to be Any due to circular import
         self._storage = storage
         if version is None:
             version = self.get_latest_version_number()
         self.version = version
-        self._metadata = {}
+        self._metadata: Any = {}
         # self.metadata is a read-only view over self._metadata
-        self.metadata = MappingProxyType(self._metadata)
-        self.response_metadata = {}
+        self.metadata: Any = MappingProxyType(self._metadata)
+        self.response_metadata: dict = {}
         self.list_store = ListStore()
         self.answer_store = AnswerStore()
         self.progress_store = ProgressStore()
+        self.submitted_at: Union[datetime, None]
```

Comment:
Optional instead of `Union` since it is one other type?.

Reply:
Optional datetime now, thanks.